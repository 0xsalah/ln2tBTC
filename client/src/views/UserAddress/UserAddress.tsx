import React, { useContext, useState, useEffect } from "react";
import "./UserAddress.css";
import Web3Context from '../../Web3Context'

export default function UserAddress() {
  const { web3 } = useContext(Web3Context)
  const [selectedAddress, setSelectedAddress] = useState<string>()

  useEffect(() => {
    setSelectedAddress(web3 === null ? undefined : (web3.currentProvider as any).selectedAddress)
    console.log('called!!!!!!', web3)
  }, [web3])

  return (
    <div className={`header__connect connect connect--${selectedAddress ? 'success' : 'no'}`}>
      <div className="connect__label">{selectedAddress ? 'Mainnet:' : 'Connect Wallet'}</div>
      {selectedAddress && <div className="connect__text">{selectedAddress.substring(0, 6)}...{selectedAddress.substring(selectedAddress.length - 4)}</div>}
      <div className="connect__status"><span></span></div>
    </div >
  );
}
