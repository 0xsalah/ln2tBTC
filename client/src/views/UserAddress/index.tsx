import React, { useContext, useState, useEffect } from "react";
import Web3Context from "../../Web3Context";
import Modal from '../Modal'

export default function UserAddress() {
  const { web3, connectWallet } = useContext(Web3Context);
  const [selectedAddress, setSelectedAddress] = useState<string>();
  const [displayNoMetamaskModal, setDisplayNoMetamaskModal] = useState(false);

  useEffect(() => {
    setSelectedAddress(
      web3 === null ? undefined : (web3.currentProvider as any).selectedAddress
    );
  }, [web3]);

  return (
    <>
      <div
        className={`header__connect connect connect--${selectedAddress ? "success" : "no"
          }`}
        onClick={() => (!selectedAddress ? connectWallet((errType) => {
          if (errType === 'NO_METAMASK')
            setDisplayNoMetamaskModal(true)
        }) : null)}
      >
        <div className="connect__label">
          {selectedAddress ? "Mainnet:" : "Connect Wallet"}
        </div>
        {selectedAddress && (
          <div className="connect__text">
            {selectedAddress.substring(0, 6) +
              "..." +
              selectedAddress.substring(selectedAddress.length - 4)}
          </div>
        )}
        <div className="connect__status">
          <span></span>
        </div>
      </div>
      <Modal
        isOpen={displayNoMetamaskModal}
        type="error"
        title={"Wallet not found"}
        buttonText={"Okay"}
        onButtonClick={() => setDisplayNoMetamaskModal(false)}
      >
        <span>
          You must have MetaMask installed to use this product, get it{" "}
          <a
            rel="noopener noreferrer"
            target="_blank"
            href="https://metamask.io/"
          >
            here
              </a>
              .
            </span>
      </Modal>
    </>
  );
}
