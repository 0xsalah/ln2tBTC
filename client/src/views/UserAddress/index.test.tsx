import React from "react";
import { render } from "@testing-library/react";
import UserAddress from ".";

test("snapshot of component rendered when no address found", () => {
  const component = render(<UserAddress />);
  expect(component.baseElement).toMatchSnapshot();
});
