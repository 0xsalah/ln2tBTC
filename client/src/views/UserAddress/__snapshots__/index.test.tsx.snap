// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered when no address found 1`] = `
<body>
  <div>
    <div
      class="header__connect connect connect--no"
    >
      <div
        class="connect__label"
      >
        Connect Wallet
      </div>
      <div
        class="connect__status"
      >
        <span />
      </div>
    </div>
  </div>
</body>
`;
